toolchain/check/eval_inst.cpp

```diff
@@ -314,6 +315,22 @@ auto EvalConstantInst(Context& context, SemIRLoc /*loc*/,
       context.constant_values().Get(inst.result_id));
 }
 
+auto EvalConstantInst(Context& context, SemIRLoc /*loc*/,
+                      SemIR::SpliceInst inst) -> ConstantEvalResult {
+  // The constant value of a SpliceInst is the constant value of the instruction
+  // being spliced. Note that `inst.inst_id` is the instruction being spliced,
+  // so we need to go through another round of obtaining the constant value in
+  // addition to the one performed by the eval infrastructure.
+  if (auto inst_value =
+          context.insts().TryGetAs<SemIR::InstValue>(inst.inst_id)) {
+    return ConstantEvalResult::Existing(
+        context.constant_values().Get(inst_value->inst_id));
+  }
+  // TODO: Consider creating a `ValueOfInst` instruction here to defer
```

Comment:
Do you mean `InstValue` or is this an instruction that hasn't been created yet?
```suggestion
  // TODO: Consider creating a `InstValue` instruction here to defer
```

Reply:
This would be a new instruction that hasn't been created yet. I've tried to make the comment a bit clearer.